models/cart-discount/src/cart-discount-value-absolute/types.ts

```diff
@@ -7,12 +7,24 @@ import type { TBuilder } from '@commercetools-test-data/core';
 export type TCartDiscountValueAbsolute = CartDiscountValueAbsolute;
 export type TCartDiscountValueAbsoluteDraft = CartDiscountValueAbsoluteDraft;
 
-export type TCartDiscountValueAbsoluteGraphql = TCartDiscountValueAbsolute & {
+export type TCartDiscountValueAbsoluteGraphql = Omit<
+  TCartDiscountValueAbsolute,
+  'applicationMode'
+> & {
   __typename: 'AbsoluteDiscountValue';
 };
 
+export type TCartDiscountValueAbsoluteCartGraphql =
+  Required<TCartDiscountValueAbsolute> & {
```

Comment:
I'm curious about the `Required` utility here. What are the benefits?

Reply:
> I'm curious about the `Required` utility here. What are the benefits?

`applicationMode` prop is optional in `TCartDiscountValueAbsolute` and mandatory in the GraphQL schema.
IMO we should have a tool in this repo to generate GraphQL types from GraphQL fragments, this way we would not need manual types like this one which could be innacurate.

Reply:
I agree. Deriving GraphQL types from Rest types doesn't seem ideal